runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -124,33 +132,77 @@ public class MqttKafkaSubscribeFactory implements MqttKafkaStreamFactory
     private final MutableDirectBuffer writeBuffer;
     private final MutableDirectBuffer extBuffer;
     private final MutableDirectBuffer subscriptionIdsBuffer;
-    private final MutableDirectBuffer retainFilterBuffer;
+    private final MutableDirectBuffer filterBuffer;
     private final BindingHandler streamFactory;
+    private final Signaler signaler;
     private final BufferPool bufferPool;
     private final LongUnaryOperator supplyInitialId;
     private final LongUnaryOperator supplyReplyId;
+    private final Supplier<Long> supplyTraceId;
     private final int mqttTypeId;
     private final int kafkaTypeId;
     private final LongFunction<MqttKafkaBindingConfig> supplyBinding;
     private final MqttKafkaHeaderHelper helper;
+    private final int reconnectDelay;
+    private final boolean bootstrapAvailable;
+
+    private int reconnectAttempt;
 
     public MqttKafkaSubscribeFactory(
         MqttKafkaConfiguration config,
         EngineContext context,
-        LongFunction<MqttKafkaBindingConfig> supplyBinding)
+        LongFunction<MqttKafkaBindingConfig> supplyBinding,
+        IntPropertyDef reconnectDelay)
     {
         this.mqttTypeId = context.supplyTypeId(MQTT_TYPE_NAME);
         this.kafkaTypeId = context.supplyTypeId(KAFKA_TYPE_NAME);
         this.writeBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
         this.extBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
         this.subscriptionIdsBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
-        this.retainFilterBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
+        this.filterBuffer = new UnsafeBuffer(new byte[context.writeBuffer().capacity()]);
         this.streamFactory = context.streamFactory();
+        this.signaler = context.signaler();
         this.bufferPool = context.bufferPool();
         this.supplyInitialId = context::supplyInitialId;
         this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
         this.supplyBinding = supplyBinding;
         this.helper = new MqttKafkaHeaderHelper();
+        this.bootstrapAvailable = config.bootstrapAvailable();
+        this.reconnectDelay = reconnectDelay.getAsInt(config);
+    }
+
+    @Override
+    public void onAttached(
+        long bindingId)
+    {
+        MqttKafkaBindingConfig binding = supplyBinding.apply(bindingId);
+        List<MqttKafkaRouteConfig> bootstrap = binding.bootstrapRoutes();
+        if (bootstrapAvailable && bootstrap != null)
```

Comment:
`bootstrap` is never null by implementation.

Perhaps use configuration to return empty list of bootstrap routes if bootstrap not available?